core/src/contracts/stablex_contract.rs

```diff
@@ -149,7 +150,11 @@ pub trait StableXContract {
     fn send_noop_transaction<'a>(
         &'a self,
         gas_price: U256,
+        nonce: U256,
     ) -> BoxFuture<'a, Result<TransactionResult>>;
+
+    /// The current nonce aka transaction_count.
+    fn get_nonce<'a>(&'a self) -> BoxFuture<'a, Result<U256>>;
```

Comment:
Can the name reflect which nonce we are fetching (something like `get_nonce_for_next_transaction` or even `get_transaction_count`)?